### ISSUE TITLE
Lower-case sequence alignments not supported?

### ISSUE THREAD

**Issue by Anto007** [citiususc/veryfasttree#20]
Hi,

I'm testing out your interesting tool and noticed that VeryFastTree (version 4.0; conda package) doesn't seem to be able to recognize lower-case a,t,g,c. I have a mafft alignment of ASVs from dada2 which worked as input on FastTree-2 but seemingly not on VeryFastTree. For the benefit of anyone encountering a similar problem, I used reformat.sh from BBMap to convert my lower-case mafft alignment to an upper-case one to feed into VeryFastTree and this seems to be running OK now (don't have the final results yet). It might be useful for future users if this uppercase sequence requirement was mentioned somewhere in the README.

**Comment by cesarpomar** [comment]
Hi,

FastTree and VeryFastTree support different input formats. While FastTree only supports formats like fasta and phylip, VeryFastTree is more versatile and supports fasta, fastq, nexus, and phylip. Each format has its own characteristics; for instance, phylip is case-insensitive, but fasta, fastq, and nexus are case-sensitive.

In the FastTree source code (http://www.microbesonline.org/fasttree/FastTree-2.1.11.c), you can observe that the 'toupper' function is only used in the phylip buffer. This is intentional because, as explained in "Why does FastTree warn about unrecognized characters?", 'U' is not the same as 'u' in case-sensitive formats. Therefore, converting everything to uppercase is not feasible if the format is case-sensitive.

If you were using the exact same file in both FastTree and VeryFastTree, please let me know, and I will verify it for you. Providing a sample file in that case would be very helpful.

In any case, thank you for using VeryFastTree.

**Comment by Anto007** [comment]
Thank you very much for your response @cesarpomar. Yes, I had tried running the exact same alignment fasta file on FastTree-2 and VeryFastTree. FastTree-2 did not appear to have a problem with the lower-case but VeryFastTree log clearly suggested it wasn't able to recognize this. Here's a bit of context as to how these input file was generated: Applied OrientNucleotides function of DECIPHER package to correctly orient all the ASV sequences from dada2 >>> Alignment of all oriented ASV sequences on MAFFT to generate the alignment fasta file that was used as input for FastTree-2 as well as VeryFastTree. I'm providing here a [download link](https://exrcsdrive.kaust.edu.sa/index.php/s/47sefRJYHRL1veX) (please use password:**veryfasttree**) to access the input file that I had used (nearly 6 GB in size) and the following are the commands that I used on FastTree-2 and VeryFastTree:
```
FastTree -nt -gamma -fastest -no2nd -spr 4 < ASVs_orient_aligned.fasta > ASVs_tree.nwk
VeryFastTree -nt -gamma -fastest -no2nd -spr 4 -threads 8 < ASVs_orient_aligned.fasta > ASVs_tree.nwk
```
As VeryFastTree did not recognize the lower-case bases in the sequence file, I used reformat from BBTools/BBMap suite via the following command-line to convert the sequences in the alignment fasta file to upper-case:
`reformat.sh in=ASVs_orient_aligned.fasta out=UC_ASVs_orient_aligned.fasta tuc ignorejunk `
I then fed this upper-case (UC_) reformatted input file (also shared on the download link) to VeryFastTree and the tree was successfully created
`VeryFastTree -nt -gamma -fastest -no2nd -spr 4 -threads 8 < UC_ASVs_orient_aligned.fasta > ASVs_tree.nwk`

**Comment by Anto007** [comment]
One more quick question to confirm: am I correct in assuming that VeryFastTree uses the same Jukes-Cantor+CAT model by default as that of FastTree-2?

**Comment by cesarpomar** [comment]
> One more quick question to confirm: am I correct in assuming that VeryFastTree uses the same Jukes-Cantor+CAT model by default as that of FastTree-2?

Yes.

> Thank you very much for your response @cesarpomar. Yes, I had tried running the exact same alignment fasta file on FastTree-2 and VeryFastTree. FastTree-2 did not appear to have a problem with the lower-case but VeryFastTree log clearly suggested it wasn't able to recognize this. Here's a bit of context as to how this input file was generated: Applied OrientNucleotides function of DECIPHER package to correctly orient all the ASV sequences from dada2 >>> Alignment of all oriented ASV sequences on MAFFT to generate the alignment fasta file that was used as input for FastTree-2 as well as VeryFastTree. I'm providing here a [download link](https://exrcsdrive.kaust.edu.sa/index.php/s/47sefRJYHRL1veX) (please use password:**veryfasttree**) to access the input file that I had used (nearly 6 GB in size) and the following are the commands that I used on FastTree-2 and VeryFastTree:
> 
> ```
> FastTree -nt -gamma -fastest -no2nd -spr 4 < ASVs_orient_aligned.fasta > ASVs_tree.nwk
> VeryFastTree -nt -gamma -fastest -no2nd -spr 4 -threads 8 < ASVs_orient_aligned.fasta > ASVs_tree.nwk
> ```
> 
> As VeryFastTree did not recognize the lower-case bases in the sequence file, I used reformat from BBTools/BBMap suite via the following command-line to convert the sequences in the alignment fasta file to upper-case: `reformat.sh in=ASVs_orient_aligned.fasta out=UC_ASVs_orient_aligned.fasta tuc ignorejunk ` I then fed this upper-case (UC_) reformatted input file (also shared on the download link) to VeryFastTree and the tree was successfully created `VeryFastTree -nt -gamma -fastest -no2nd -spr 4 -threads 8 < UC_ASVs_orient_aligned.fasta > ASVs_tree.nwk`

Fixed, until release 4.0.4 you can still use BBTools/BBMap or manually compile the code from the master branch.
